lib/solver.cpp

```diff
@@ -360,7 +360,7 @@ namespace quda {
 
     // check the L2 relative residual norm if necessary
     if ((param.residual_type & QUDA_L2_RELATIVE_RESIDUAL) || (param.residual_type & QUDA_L2_ABSOLUTE_RESIDUAL)) {
-      if (std::isnan(r2) || std::isinf(r2)) errorQuda("Solver appears to have diverged with residual %e", hq2);
+      if (std::isnan(r2) || std::isinf(r2)) errorQuda("Solver appears to have diverged with residual %9.6e", hq2);
```

Comment:
As this line was touched anyway: Can we print r2 here? @maddyscientist  has already fixed that in the kernels branch.

Reply:
Good call @mathiaswagner: we want this fix to be in develop.

Reply:
FIxed.